wpilibj/src/main/java/edu/wpi/first/wpilibj/ADIS16448_IMU.java

```diff
@@ -1042,16 +1084,36 @@ public IMUAxis getYawAxis() {
     return m_yaw_axis;
   }
 
+  /**
+   * Returns the raw (un-offset) orientation of the device as a Rotation3d
+   *
+   * @return Rotation3d representing the device orientation
+   */
+  private synchronized Rotation3d getGyroOrientation() {
+    Rotation3d m_orientation;
+    if (m_simGyroAngleX != null && m_simGyroAngleY != null && m_simGyroAngleZ != null) {
+      m_orientation =
+          new Rotation3d(
+              Units.degreesToRadians(m_simGyroAngleX.get()),
+              Units.degreesToRadians(m_simGyroAngleY.get()),
+              Units.degreesToRadians(m_simGyroAngleZ.get()));
+    } else {
+      m_orientation =
+          new Rotation3d(
+              Units.degreesToRadians(m_integ_gyro_angle_x),
+              Units.degreesToRadians(m_integ_gyro_angle_y),
+              Units.degreesToRadians(m_integ_gyro_angle_z));
+    }
+    return m_orientation;
+  }
+
   /**
    * Returns the accumulated gyro angle in the X axis in degrees.
    *
    * @return The accumulated gyro angle in the X axis in degrees.
    */
   public synchronized double getGyroAngleX() {
-    if (m_simGyroAngleX != null) {
-      return m_simGyroAngleX.get();
-    }
-    return m_integ_gyro_angle_x;
+    return Units.radiansToDegrees(getGyroOrientation().minus(m_angleOffset).getX());
```

Comment:
Nit: Instead of setting the offset to the negative of the provided angle and then subtracting, it'd be simpler to directly store the provided angle to reset to and then add it in the getters.